easydeasy_rest/api/views.py

```diff
@@ -13,6 +16,36 @@ class DiseaseViewSet(ModelViewSet):
     queryset = Disease.objects.all()
     serializer_class = DiseaseSerializer
 
+    @action(detail=False, url_path='(?P<url_disease_id>[0-9]+)/latest_articles')
+    def latest_disease_articles(self, request, url_disease_id):
+        """ returns the list of newest articles.
+
+            (?P<url_disease_id>[0-9]+) in the decorator forces to only receive ints for url_disease_id.
+            """
+
+        # Converts string url to int
+        disease_id = int(url_disease_id)
+
+        article_queryset = Article.objects.filter(diseases__id=disease_id)
+
+        # A pagination class used specifically to page the latest articles based on there dates
+        class LatestArticlesSetPagination(PageNumberPagination):
+            page_size = 10
+
+        pagination = LatestArticlesSetPagination()
+
+        page = pagination.paginate_queryset(article_queryset, request)
+
+        # Checks if we even got a page
+        if page:
+            serializer = ArticleSerializer(page, many=True, context={'request': request})
+            return pagination.get_paginated_response(serializer.data)
+
+        # Creates the serializer making sure that we get the correct url path for the images
+        serializer = ArticleSerializer(article_queryset, many=True, context={'request': request})
+
+        return Response(serializer.data)
+
 
```

Comment:
We have 2 return statements, It's also confusing to understand exactly what is happening. 

So if there's a page we serialize the page and return it.
But if there's not a page, we just return the latest articles for that disease?